### PR TITLE
treewide: update urls to https using HTTPS Everywhere

### PR DESCRIPTION
**Pull request by J0WI**
Maintainer: --
Compile tested: --
Run tested: --

_No functional changes._

Description:
This updates the urls in packages from `http` to `https` where possible.

The changes can be reproduced by using [this script](https://github.com/EFForg/https-everywhere/tree/573a16793923d0ce3ea5b7c79fd4463ae8c94867/utils/rewriter), applying [these patches](https://github.com/EFForg/https-everywhere/issues/15750), removing [this line](https://github.com/EFForg/https-everywhere/blob/573a16793923d0ce3ea5b7c79fd4463ae8c94867/utils/rewriter/rewriter.js#L78) and using `Makefile` [here](https://github.com/EFForg/https-everywhere/blob/573a16793923d0ce3ea5b7c79fd4463ae8c94867/utils/rewriter/rewriter.js#L37-L40). `PKG_RELEASE` has been bumped with `perl -i -pe 's/PKG_RELEASE:=\K\d+/$&+1/e'`.

### CODE REVIEW

**Comment by neheb** [comment]
I don't think most of these require bumping the package versions as there are no functional differences.

**Comment by J0WI** [comment]
@neheb should I revert the bump for all packages?

**Comment by neheb** [comment]
Probably. HTTP vs. HTTPS doesn't matter much for URLs when building since the hash is verified afterwards. The other URLs are cosmetic and don't really impact anything.

**Comment by thess** [comment]
Your signed-off-by tag does not contain your real name (sorry, we don't accept PRs without knowing who you are. See: [Sign your work](https://openwrt.org/docs/guide-developer/submittingpatches-tomerge?s[]=sign#sign_your_work)
>Author name (J0WI) need to be your real name 'firstname lastname'

Have all of these downloads been verified? You don't mention ANY testing.
I vote NO on these changes as they are really unnecessary. Just my 2-cents.

File: libs/libtheora/Makefile [hunk 1]
```diff
@@ -28,7 +28,7 @@ define Package/libtheora
   SECTION:=libs
   CATEGORY:=Libraries
   TITLE:=libtheora
-  URL:=http://xiph.org/theora/
+  URL:=https://xiph.org/theora/
```
Comment (diizzyy) [inline libs/libtheora/Makefile]: Proper URL --> https://www.theora.org/

File: net/fakeidentd/Makefile [hunk 2]
```diff
@@ -12,7 +12,7 @@ PKG_VERSION:=2.6
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).c
-PKG_SOURCE_URL:=http://distfiles.gentoo.org/distfiles/
+PKG_SOURCE_URL:=https://gentoo.osuosl.org/distfiles/
```
Comment (diizzyy) [inline net/fakeidentd/Makefile]: NAK, updating URLs are fine but changing should be a separate PR

File: libs/alsa-lib/Makefile [hunk 3]
```diff
@@ -13,7 +13,7 @@ PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
 PKG_SOURCE_URL:=ftp://ftp.alsa-project.org/pub/lib/ \
-		http://distfiles.gentoo.org/distfiles/
+		https://gentoo.osuosl.org/distfiles/
```
Comment (diizzyy) [inline libs/alsa-lib/Makefile]: NAK, updating URLs are fine but changing should be a separate PR

File: utils/yunbridge/Makefile [hunk 4]
```diff
@@ -25,7 +25,7 @@ define Package/yunbridge
   SECTION:=utils
   CATEGORY:=Utilities
   TITLE:=Arduino YUN bridge library
-  URL:=http://arduino.cc/
+  URL:=https://arduino.cc/
```
Comment (diizzyy) [inline utils/yunbridge/Makefile]: Proper URL --> https://www.arduino.cc/

File: sound/alsa-utils/Makefile [hunk 5]
```diff
@@ -13,7 +13,7 @@ PKG_RELEASE:=2
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
 PKG_SOURCE_URL:=ftp://ftp.alsa-project.org/pub/utils/ \
-		http://distfiles.gentoo.org/distfiles/
+		https://gentoo.osuosl.org/distfiles/
```
Comment (diizzyy) [inline sound/alsa-utils/Makefile]: NAK, updating URLs are fine but changing should be a separate PR

**Comment by diizzyy** [comment]
@thess 
I did have a look at most of them and they seem fine but I agree that it should at least be tested.

**Comment by J0WI** [comment]
What test procedure do you propose for this?

**Comment by hnyman** [comment]
There are currently conflicts, so this does not merge nicely, any more. Needs work in any case.

I feel that there are too many changes in the PR, as discussed already above.

* One easy class is changing the pure "home page" link URL fields, which is principally ok, as it is just a link to the upstream home page. (assuming that you have tested all those https links with browser or curl or ...)
* Second class is changing the source PKG_SOURCE_URL fields, which need build system download testing, like suggested above by luizluca
* Third, a more difficult class, is changing the download site totally, like for fakeidentd. That again requires more reasoning.

**Comment by J0WI** [comment]
Thanks for you feedback!

> Have all of these downloads been verified? You don't mention ANY testing.

The redirects are made based on manually defined rules. At HTTPS Everywhere we review all rules and automated tests check each url. Redirecting to different urls are only made if there is a very good reason to do this.

> Your signed-off-by tag does not contain your real name. Sorry, we don't accept PRs without knowing who you are.

I'm sorry, but this is a blocker for me. It would be great if someone picks this task up (or maybe a reasonable subset of it). All necessary steps and scripts are [provided](https://github.com/openwrt/packages/pull/7147#issue-220903093). I also welcome any feedback and suggestions for HTTPS Everywhere rulesets.
